grails-app/services/org/pih/warehouse/data/ProductSupplierDataService.groovy

```diff
@@ -141,9 +141,27 @@ class ProductSupplierDataService {
     }
 
     def getOrCreateNew(Map params) {
-        def productSupplier = params.productSupplier ? ProductSupplier.get(params.productSupplier) : null
+        def productSupplier
+        if (params.productSupplier) {
+            productSupplier = params.productSupplier ? ProductSupplier.get(params.productSupplier) : null
+        } else {
+            productSupplier = ProductSupplier.createCriteria().list {
+                eq("product", params.product)
+                if (params.supplier && params.supplierCode && params.product) {
+                    eq("supplier", params.supplier)
+                    eq("supplierCode", params.supplierCode.replaceAll('[,-]',''))
```

Comment:
Looks like you missed whitespace.

Reply:
In the ticket there was a request for `dash`, `space`, and `dot`? Here is `comma` and `dash`